src/reason_pprint_ast.ml

```diff
@@ -630,9 +630,6 @@ let rules = [
     (TokenPrecedence, (fun s -> (Right, s = "asr")));
   ];
   [
-    (TokenPrecedence, (fun s -> (Left, s.[0] == '#')));
```

Comment:
So `!`, `PREFIXOP` are parsed with precedence `below_DOT_AND_SHARP`, which is, well - below dot and sharp/`SHARPOP`. That `below_DOT_AND_SHARP` is what `isSimplePrefixToken` tests. So that means the `SHARPOP` identifiers need to be considered higher precedence than the test for `isSimplePrefixToken`.

The fact that the tests pass even with the incorrect ordering in `rules`, means we're missing a test case!

I might suggest adding the failing test case, then changing the order to what I suggested, and you would then see it pass!

The incorrect precedence ranking is between any `SHARPOP` and anything that returns true for `isSimplePrefixToken ` (`PREFIXOP`, `!` etc).

So some good test cases would be:

```rust
/* These should print the same */
let res = (! (x##y));   /* AST */
let res = !x##y;        /* Minimum parens */

/* These should print the same */
let res = (!(z##q)##a);   /* AST */
let res = !z##q##a;   /* Min parens */

/*
 * And then copy/paste those tests, but with another `isSimplePrefixToken`
 * instead of `!` (for example `?!!` should have the same precedence as !)
 */
```